examples/1bitsy/tft-st7735/st7735lcd.c

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright (C) 2017 Mark Osborne @BecomingMaker
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "../common/button_boot.h"
+#include "systick.h"
+
+#include <libopencm3/stm32/rcc.h>
+#include <libopencm3/stm32/gpio.h>
+#include <libopencm3/stm32/spi.h>
+
+#define MY_CLOCK (rcc_hse_25mhz_3v3[RCC_CLOCK_3V3_168MHZ])
+
+// 1bitsy pins connected to TFT
+#define TFT_SCK GPIO10
+#define TFT_MOSI GPIO15
+#define TFT_NSS GPIO12
+#define TFT_RESET GPIO11
+#define TFT_DC GPIO1
+
+// TFT Commands
+#define SWRESET 0x01    // software reset
+#define SLPOUT  0x11    // sleep out
+#define DISPOFF 0x28    // display off
+#define DISPON  0x29    // display on
+#define CASET   0x2A    // column address set
+#define RASET   0x2B    // row address set
+#define RAMWR   0x2C    // RAM write
+#define MADCTL  0x36    // axis control
+#define COLMOD  0x3A    // color mode
+
+// 1.8" TFT display constants
+#define XSIZE   128
+#define YSIZE   160
+#define XMAX    XSIZE-1
+#define YMAX    YSIZE-1
+
+// Color constants
+#define BLACK   0x0000
+#define BLUE    0x001F
+#define RED     0xF800
+#define GREEN   0x0400
+#define LIME    0x07E0
+#define CYAN    0x07FF
+#define MAGENTA 0xF81F
+#define YELLOW  0xFFE0
+#define WHITE   0xFFFF
+
+static void systick_setup(void)
+{
+    rcc_clock_setup_hse_3v3(&MY_CLOCK);
+    setup_systick(MY_CLOCK.ahb_frequency);
```

Comment:
You can just pass `rcc_ahb_frequency` to setup_systick. `rcc_clock_setup_hse_3v3` sets it as a global variable so you don't have to carry along system clock frequency yourself. Reference: https://github.com/libopencm3/libopencm3/blob/master/lib/stm32/f4/rcc.c#L723